test/tests/functional/pbs_hook_execjob_end.py

```diff
@@ -0,0 +1,260 @@
+                          "e.accept()\n")
+
+    def test_execjob_end_non_blocking(self):
+        """
+        Test to make sure that mom is unblocked and running
+        exechost_periodic hook while it's child process is executing
+        execjob_end hook.
+        """
+        hook_name = "execjob_end_logmsg"
+        self.server.create_import_hook(hook_name, self.attr, self.hook_body)
+        hook_name = "exechost_periodic_logmsg"
+        hook_body = ("import pbs\n"
+                     "e = pbs.event()\n"
+                     "pbs.logmsg(pbs.LOG_DEBUG, \
+                                 'executed exechost_periodic hook')\n"
+                     "e.accept()\n")
+        attr = {'event': 'exechost_periodic', 'freq': '3', 'enabled': 'True'}
+        j = Job(TEST_USER)
+        j.set_sleep_time(1)
+        self.server.create_import_hook(hook_name, attr, hook_body)
+        jid = self.server.submit(j)
+        # need to verify hook messages in the below mentioned order to
+        # confirm mom is not blocked on execjob_end hook execution.
+        # The order is verified with the use of starttime and endtime
+        # parameters.
+        (_, str1) = self.mom.log_match("Job;%s;executed execjob_end hook" %
+                                       jid, n=100, max_attempts=10, interval=2)
+        date_time1 = str1.split(";")[0]
+        epoch1 = int(time.mktime(time.strptime(
+            date_time1, '%m/%d/%Y %H:%M:%S')))
+        # following message should be logged while execjob_end hook is in sleep
+        (_, str1) = self.mom.log_match("executed exechost_periodic hook",
+                                       starttime=epoch1-1, endtime=epoch1+10,
+                                       n=100, max_attempts=10, interval=1)
+        date_time2 = str1.split(";")[0]
+        epoch2 = int(time.mktime(time.strptime(
+            date_time2, '%m/%d/%Y %H:%M:%S')))
+        (_, str1) = self.mom.log_match(
+            "Job;%s;execjob_end hook ended" %
+            jid, starttime=epoch2-1, n=100,
+            max_attempts=10, interval=2)
+        date_time3 = str1.split(";")[0]
+        self.logger.info(
+            "execjob_end hook executed at: %s,"
+            "exechost_periodic at: %s and execjob_end hook ended at: %s" %
+            (date_time1, date_time2, date_time3))
+
+    def test_execjob_end_hook_order_and_reject(self):
+        """
+        Test with multiple execjob_end hooks having different order
+        with one of the hooks rejecting the job.
+        """
+        hook_name1 = "execjob_end_logmsg1"
+        hook_body = ("import pbs\n"
+                     "e = pbs.event()\n"
+                     "pbs.logjobmsg(e.job.id, \
+                                  'executed %s hook' % e.hook_name)\n"
+                     "e.accept()\n")
+        attr = {'event': 'execjob_end', 'order': '1', 'enabled': 'True'}
+        self.server.create_import_hook(hook_name1, attr, hook_body)
```

Comment:
The variable names are inconsistent. `hook_name1` and `hook_body` are one hook, `hook_name` and `hook_body1` are another hook.
Maybe have `hook_body_reject` and `hook_body_accept`?